### ISSUE TITLE
Cannot enable Major version roll-forward for 2.x apps

### ISSUE THREAD

**Issue by richlander** [dotnet/cli#12257]
## Steps to reproduce

I attempted to enable major-version roll-forward for the [dotnetsay .NET Core tool](https://github.com/dotnet/core/blob/master/samples/dotnetsay/dotnetsay.csproj).

I added `<RollForward>LatestMajor</RollForward>` within the `PropertyGroup`. I get this error:

```console
C:\git\core\samples\dotnetsay>dotnet build
Microsoft (R) Build Engine version 16.3.0-preview-19377-01+dd8019d9e for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 131.58 ms for C:\git\core\samples\dotnetsay\dotnetsay.csproj.
  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
C:\Program Files\dotnet\sdk\3.0.100-preview8-013656\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(221,5): error NETSDK1103: RollForward setting is only supported on .NET Core 3.0 or higher. [C:\git\core\samples\dotnetsay\dotnetsay.csproj]

Build FAILED.
```

## Expected  behavior

I understand why RollForward isn't enabled in general for 2.x since the feature is new in 3.0. We should special-case the `LatestMajor` value to enable it to be used. It is an important scenario for enabling 3.0 users (on 3.x-only machines) to use 2.x tools. 

The only other option to enable 2.x tools to run on 3.x is to multi-target. I'm not a fan of that being the only option (in the case you are only doing it for this scenario and not taking advantage of 3.x APIs).

## Timeframe

Please consider for 3.0.

/cc @vitek-karas @livarcocc @KathleenDollard

**Comment by vitek-karas** [comment]
Technically it is possible to do this via `.runtimeconfig.template` but I agree that's not very nice or discoverable.

**Comment by richlander** [comment]
That's what I'm going to do in this particular case (to unblock myself) but I still want the fix considered. It's solely a question of whether this feature (major version roll-forward) is important for the 2.x to 3.x boundary only starting with 3.x to 5.x. My opinion should be pretty clear at this point.

**Comment by vitek-karas** [comment]
I think we should enable it for both `Major` and `LatestMajor` - the global tools may want `LatestMajor` but I think even `Major` would make sense as well.

**Comment by richlander** [comment]
Relevant tweet that formed much of the motivation for this feature: https://twitter.com/KathleenDollard/status/1079811275641696256

**Comment by livarcocc** [comment]
@vitek-karas are you going to do a bar check on this?

**Comment by richlander** [comment]
@vitek-karas sent mail, so yes.

**Comment by richlander** [comment]
Ya, we're going to take this fix. It's approved.

**Comment by vitek-karas** [comment]
Fixed by dotnet/sdk#3542
